### DISCUSSION TITLE
How to add Grid and snapping

### DISCUSSION THREAD

**Discussion by Udhayan26** [fabricjs/fabric.js#7810] [General]
Hi all,
Ref: (http://jsfiddle.net/ej2hrqm8/)

In my canvas drawings, patterns are shapes. I used the above reference script, but snap and grid using external objects did not work. How do I fix this? I also need snap and grid optional types.

Thankyou

**Comment by ShaMan123** [comment]
Very unclear what you need/want
What are external objects?
Help others help you.
The example you gave seems fine.

**Reply by Udhayan26** [reply]
Thanks for your reply.

I need snap and grid for fabricjs.
